mart/attack/adversary.py

```diff
@@ -24,31 +24,32 @@ class AdversaryCallbackHookMixin(Callback):
 
     callbacks = {}
 
-    def on_run_start(self, adversary, input, target, model, **kwargs) -> None:
+    def on_run_start(self, **kwargs) -> None:
         """Prepare the attack loop state."""
         for _name, callback in self.callbacks.items():
             # FIXME: Skip incomplete callback instance.
-            callback.on_run_start(adversary, input, target, model, **kwargs)
+            # Give access of self to callbacks by `adversary=self`.
+            callback.on_run_start(adversary=self, **kwargs)
```

Comment:
This is a big logic change and a bad thing to do. The callee should supply `adversary=self` since this is a mixin.

Reply:
OK. No reference to self in mixin.